registry/network/helpers_test.go

```
@@ -0,0 +1,45 @@
+package network
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestDetermineIPVersion tests the determineIPVersion function.
+func TestDetermineIPVersion(t *testing.T) {
+	nr := &NetworkRegistry{}
+
+	// Test case: IPv4 address
+	ipv4 := net.ParseIP("192.168.0.1")
+	assert.Equal(t, 4, nr.determineIPVersion(ipv4))
+
+	// Test case: IPv6 address
+	ipv6 := net.ParseIP("2001:db8::1")
+	assert.Equal(t, 6, nr.determineIPVersion(ipv6))
+
+	// Test case: IPv4-mapped IPv6 address
+	ipv4MappedIPv6 := net.ParseIP("::ffff:192.168.0.1")
+	assert.Equal(t, 4, nr.determineIPVersion(ipv4MappedIPv6))
+}
+
+// TestCalculateLastIP tests the calculateLastIP function.
+func TestCalculateLastIP(t *testing.T) {
+	nr := &NetworkRegistry{}
+
+	// Test case: IPv4 CIDR
+	_, cidrIPv4, _ := net.ParseCIDR("192.168.0.0/24")
```

Here an everywhere, you shouldn't ignore the error returned by ParseCIDR